main/src/ca/uwaterloo/flix/util/collection/Chain.scala

```diff
@@ -100,9 +92,13 @@ sealed trait Chain[+A] {
     * Applies `f` this every element in `this`.
     */
   final def foreach(f: A => Unit): Unit = this match {
-    case Chain.Empty => ()
-    case Chain.Link(l, r) => l.foreach(f); r.foreach(f)
-    case Chain.Proxy(xs) => xs.foreach(f)
+    // N.B.: We have to use reflection to avoid
```

Comment:
Does this quality as reflection?

Reply:
No, its not reflection. Just instanceof.

Reply:
- Update comment (actually just remove, this is standard scala code)

Reply:
Why not update it to
```
    // N.B.: We must use type annotations
    // infinite recursion when pattern matching
    // since it calls the equals method which
    // depends on toList.
```

Reply:
We must "match on the type" -
same text